exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKitExtension.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.testkit;
+
+import com.google.common.collect.ImmutableSet;
+import java.lang.reflect.Method;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ExtensionContext.Store;
+import org.junit.jupiter.api.extension.ExtensionContext.Store.CloseableResource;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+/**
+ * Extension that injects TestKit into service tests and destroys afterwards. Register this
+ * extension with TestKit builder and a TestKit will be injected as a parameter, instantiated from
+ * given builder. Example usage:
+ *
+ * <pre><code>
+ * &#64;RegisterExtension
```

Comment:
I see neither examples of creating `TestKit` in a declarative way (using `@ExtendWith`) or a Javadoc describing why it's not possible/forbidden.

Reply:
It's not possible to use `@ExtendWith` because that way we can't pass a TestKit builder as an argument to extension constructor. Should I specify that in Javadoc?

Reply:
Since we provide examples of usage right in the Javaddocs (and, later, on the web), I don't think it is necessary to say that "You can't register it with ExtendWith". If users try and it does not work — we can reasonably assume they will get to the (any) docs.